MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/VanGenuchtenCapillaryPressureSaturation.h

```diff
@@ -78,6 +83,71 @@ class VanGenuchtenCapillaryPressureSaturation final
 private:
     const double _pb;  ///< Entry pressure.
     const double _m;   ///< Exponent m, m in [0,1]. n=1/(1-m).
+    bool _has_Regularized;
+    const double xi = 1e-5;
+
+private:
+    /**
+    * regularized van Genuchten capillary pressure-saturation Model
+    */
+    double getPc_bar_vG_Sg(double Sg) const
+    {
+        double const S_gr = 0.0;
+        double const S_bar = getS_bar(Sg);
+        double const Pc_bar_vG =
+            getPc_vG_Sg(S_bar) -
+            getPc_vG_Sg(
+                S_gr +
+                (1 - S_gr - CapillaryPressureSaturation::_saturation_r) * xi /
+                    2);
+        return Pc_bar_vG;
+    }
+    /**
+    * regularized van Genuchten capillary pressure-saturation Model
+    */
+    double getS_bar(double Sg) const
+    {
+        double const S_gr = 0.0;
+        return S_gr + (1 - xi) * (Sg - S_gr) +
+               0.5 * xi *
+                   (1 - S_gr - CapillaryPressureSaturation::_saturation_r);
+    }
+    /**
+    *  van Genuchten capillary pressure-saturation Model
+    */
+    double getPc_vG_Sg(double Sg) const
+    {
+        double const S_gr = 0.0;
+        double const S_le =
+            (1 - Sg - CapillaryPressureSaturation::_saturation_r) /
+            (1 - S_gr - CapillaryPressureSaturation::_saturation_r);
+        return _pb * std::pow(std::pow(S_le, (-1.0 / _m)) - 1.0, 1.0 - _m);
+    }
+    /**
+    * derivative dPCdS based on regularized van Genuchten capillary
+    * pressure-saturation Model
+    */
+    double get_dPCdS_vG_bar(double Sg) const
+    {
+        double S_bar = 0.0;
+        S_bar = getS_bar(Sg);
```

Comment:
You can directly initialize S_bar with the required value. Make it const afterwards.